Tempo/Controllers/PostHistoryTableViewController.swift

```diff
@@ -46,33 +49,101 @@ class PostHistoryTableViewController: PlayerTableViewController, PostViewDelegat
     override func viewDidAppear(animated: Bool) {
 		super.viewDidAppear(animated)
 		
-		if index != nil {
-			let selectedRow = NSIndexPath(forRow: index!, inSection: 0)
+		if sectionIndex != nil {
+			let selectedRow = NSIndexPath(forRow: 0, inSection: sectionIndex!)
 			tableView.scrollToRowAtIndexPath(selectedRow, atScrollPosition: UITableViewScrollPosition.Top, animated: true)
 		}
     }
 	
-    // TableView Methods
+	private func convertDate(date: String) -> String {
+		let year = date.substringToIndex(date.startIndex.advancedBy(4))
+		let monthDay = date.substringFromIndex(date.startIndex.advancedBy(5))
+		let editedDate = monthDay + "/" + year.substringFromIndex(year.startIndex.advancedBy(2))
+		let d = editedDate.stringByReplacingOccurrencesOfString("/", withString: ".")
+		return d
+	}
 	
+	// Get the absolute index path of cell
+	private func absoluteIndex(indexPath: NSIndexPath) -> Int {
+		var absoluteIndex = indexPath.row
+		if indexPath.section > 0 {
+			for s in 0...indexPath.section-1 {
+				absoluteIndex = absoluteIndex + postedDatesDict[postedDatesSections[s]]!
+			}
+		}
+		return absoluteIndex
+	}
+	
+	// Filter posted dates into dictionary of key: date, value: date_count
+	func filterPostedDatesToSections(dates: [NSDate]) {
+		// Clear section dictionary
+		postedDatesDict = [String: Int]()
+		postedDatesSections = []
+		// Create new dictionary
+		for d in dates {
+			let date = d.yearMonthDay()
+			if let count = postedDatesDict[date] {
+				postedDatesDict[date] = count + 1
+			} else {
+				postedDatesDict[date] = 1
+				postedDatesSections.append(date)
+			}
+		}
+	}
+	
+    // TableView Methods
+
     override func tableView(tableView: UITableView, cellForRowAtIndexPath indexPath: NSIndexPath) -> UITableViewCell {
 		let cell = tableView.dequeueReusableCellWithIdentifier("FeedCell", forIndexPath: indexPath) as! FeedTableViewCell
 		
 		cell.postView.type = .History
 		let posts = searchController.active ? filteredPosts : self.posts
-		cell.postView.post = posts[indexPath.row]
+		cell.postView.post = posts[absoluteIndex(indexPath)]
 		cell.postView.delegate = self
 		cell.postView.post?.player.prepareToPlay()
 		
-        let date = NSDateFormatter.simpleDateFormatter.stringFromDate(postedDates[indexPath.row])
-		cell.postView.dateLabel!.text = "\(date)"
+//        let date = convertDate(postedDatesSections[indexPath.section])
```

Comment:
Lol yeah, no. But the overall design of the feed cell is gonna change too so it will be changed again once we implement the design changes